addons/udes_advanced_packaging/models/stock_move_line.py

```diff
@@ -0,0 +1,14 @@
+from odoo import models
+from collections import defaultdict
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _get_all_products_quantities(self):
```

Comment:
If this is called on two different pickings, one with UoM of things in 10's the other in things in 1, and they have prouct_uom_qty = 10 each.
This may not be as useful, because if would say {apple: 20} but in terms of the actual quantity of apples (if they are in units) it is 110.
I think this would only be done on ML in the same picking (as a guess)
So we could:
a. Use product_qty so then it is in the UoM of the product (would give 110 here)
b. raise error if ml with the same products have differnent UoMs (possibly safer)
c. Add to the doc string

Optionally, add a kwarg to return it in the UoM of the move, or the product.

Not sure how it is used, haven't checked but just something to consider.

Reply:
I ported this behaviour over, are you suggesting this as an improvement to how current UDES works?

Reply:
As discussed elsewhere, let's add TODOs referencing issue 962.